src/compatibility.h

```diff
@@ -0,0 +1,47 @@
+#ifndef CATA_COMPATIBILITY_H
+#define CATA_COMPATIBILITY_H
+
+//--------------------------------------------------------------------------------------------------
+// HACK:
+// std::to_string is broken on MinGW (as of 13/01/2015), but is fixed in MinGW-w64 gcc 4.8.
+// See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=52015
+// This is a minimal workaround pending the issue begin solved (if ever).
+// For proper linking these must be declared inline here, or moved to a cpp file.
+//--------------------------------------------------------------------------------------------------
+#include <string>
+
+#define CATA_GCC_VER (__GNUC__ * 10000) + (__GNUC_MINOR__ * 100) + (__GNUC_PATCHLEVEL__)
+
+#if defined(__MINGW32__) && !defined(__MINGW64__)
+#   define CATA_NO_CPP11_STRING_CONVERSIONS
+#elif defined(__MINGW64__) && (CATA_GCC_VER < 40800)
+#   define CATA_NO_CPP11_STRING_CONVERSIONS
+#endif
+
+#if defined(CATA_NO_CPP11_STRING_CONVERSIONS)
+#include <cstdio>
+#include <limits>
+
+namespace std {
```

Comment:
Uh... are you sure it's permitted to mess around in namespace std like this? I seem to recall having read something recently that suggested it shouldn't work, but I'm not a C++ person.